From 0ba80d6d558ec3230a38e01cff91a5a4b326eba0 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Thu, 16 Apr 2026 13:44:59 +0800
Subject: [PATCH 1/7] Include MindTheGapps

---
 lineage_dodge.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lineage_dodge.mk b/lineage_dodge.mk
index 17f87eb..76daf1e 100644
--- a/lineage_dodge.mk
+++ b/lineage_dodge.mk
@@ -29,3 +29,6 @@ PRODUCT_BUILD_PROP_OVERRIDES += \
     DeviceProduct=CPH2653 \
     SystemDevice=OP5D55L1 \
     SystemName=CPH2653
+
+
+$(call inherit-product, vendor/gapps/arm64/arm64-vendor.mk)
-- 
2.54.0

